tests/qemu-tests/tests.sh

```diff
@@ -216,13 +209,19 @@ test_system() {
     pkill qemu-system
 }
 
-for platform in "${PLATFORMS[@]}"; do
-    read -r arch kernel_type qemu_args <<< "$platform"
+for vmlinux in "${VMLINUX_LIST[@]}"; do
 
-    ARCH="$arch"
-    KERNEL_TYPE="$kernel_type"
-    QEMU_ARGS=($qemu_args)
-    VMLINUX="${IMAGE_DIR}/vmlinux-${KERNEL_TYPE}-${ARCH}"
+    VMLINUX="${IMAGE_DIR}/${vmlinux}"
+    KERNEL=$(echo "${vmlinux}" | sed "s/vmlinux-//")
+    # extract architecture as last dash-separated group of the kernels name
+    ARCH="${KERNEL##*-}"
+    QEMU_ARGS=()
 
     test_system
```

Comment:
Maybe we should pass the kernel, arch, and any extra arguments as arguments to `test_system` instead of global variables?

Reply:
I thought about doing that but then those arguments have to be passed further down to functions called from withing `test_system` and when I tried it I found that it made the code uglier. But if you still think so, I can change it back.

Reply:
I still think passing this as function arguments would be better, but would be worth getting another opinion. @disconnect3d?

Reply:
No worries, if you think so it's okay. I just pushed a new commit for this.

Reply:
tbh with bash I just don't like using more than one or maximum two arguments as they tend to be confusing/error prone when editing the code later on. I always have to scroll back and forth between the definition and the reference and check that i didn't miss any hard-to-spot function argument e.g. `$@` or similar somewhere down the line which messes everything up :)
At some point the logic should probably be ported to e.g. python where all function arguments are clearly visible in the function signature